tasks/tarball.yml

```diff
@@ -5,9 +5,31 @@
 - name: "Ensure the zookeeper dir exists at {{zookeeper_dir}}"
   file: path={{zookeeper_dir}} state=directory
 
+# If the file exists already, the S3 call can experience errors
+- name: check whether the tarball is downloaded already
+  stat: path={{zookeeper_tarball_dir}}/zookeeper-{{zookeeper_version}}.tar.gz
+  register: zookeeper_tarball_script
+
+- name: Download zookeeper from S3
+  s3:
+    mode: get
+    bucket: "{{zookeeper_archive_s3_bucket}}"
+    object: "{{zookeeper_archive_s3_object}}"
+    dest: "{{zookeeper_tarball_dir}}/zookeeper-{{zookeeper_version}}.tar.gz"
+  register: zookeeper_s3_tarball_downloaded
+  tags: bootstrap
+  when: >
+    zookeeper_archive_s3_bucket is defined and
+    zookeeper_archive_s3_object is defined and
+    not zookeeper_tarball_script.stat.exists
```

Comment:
We have to make sure all vars are defined, and that the tarball hasn't already been downloaded.  The S3 directive will fail in some cases if the file exists.